4.15.1 broken scope: ZonedDateTime and others no longer available unless specifically imported

Just upgraded my library from 4.14 to 4.17 and all my scripts using ZonedDateTime broke because now I need to issue an explicit java_import Java::JavaTime::ZonedDateTime inside the rule / method. Importing it globally at the top of the script didn't work.

I am a bit confused with this. In my production system, in one test.rb it works fine without importing ZonedDateTime both outside a rule, and inside a rule's run block. So it seems to behave the same as before.

Yet in another rule file of the same system, it gave me an error if I didn't import ZonedDateTime.

```
01:13:07.445 [ERROR] [yobj.OpenHAB.DSL.Rules.AutomationRule] - undefined method `now' for nil:NilClass (NoMethodError)
```

If I find more info I'll reopen this issue.

It all very weird. I also see:
```
01:14:59.798 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/xxxx.rb': Error during evaluation of Ruby in org/jruby/RubyClass.java at line 879: (TypeError) no public constructors for Java::JavaTime::ZonedDateTime
```

I cannot explain this. With 4.15.0 this problem doesn't happen. With 4.15.1, it happens. But if I changed the return value of `OpenHAB::DSL::Rules::rule` method from returning `rule` to returning `config`, the problem didn't happen. Changed it back to returning `rule`, and the problem happened again.

I tried changing my openhab version, from 3.1.0 to 3.2M4 to 3.2 snapshot, this doesn't make a difference.

A simple way to reproduce it on my system:

conf/automation/jsr223/ruby/personal/test1.rb
```
require 'openhab'

rule 'T1' do
  received_command DimmerItem1
  run { logger.error("Test1: #{ZonedDateTime.now}") }
end
```

Then send a command to DimmerItem1 from the console (or from another rules file)

There are errors if I sent the command from within the same rules file (i.e. from within test1.rb)

The error:
```
20:48:59.931 [ERROR] [yobj.OpenHAB.DSL.Rules.AutomationRule] - undefined method `now' for nil:NilClass (NoMethodError)
In rule: T1
/openhab/conf/automation/jsr223/ruby/test1.rb:5:in `block in <main>'
```

What's strange is I cannot reproduce this using cucumber test (running on a different machine) even on 4.19.

--- a/lib/openhab/dsl/rules/rule.rb
+++ b/lib/openhab/dsl/rules/rule.rb
@@ -38,6 +38,7 @@ module OpenHAB
         config.guard = Guard::Guard.new(only_if: config.only_if, not_if: config.not_if)
         logger.trace { config.inspect }
         process_rule_config(config)
+        nil # Must return something other than the rule object. See https://github.com/boc-tothefuture/openhab-jruby/issues/438
       rescue StandardError => e
         re_raise_with_backtrace(e)
       end

